oracles/oracle_ws_api.md

```diff
@@ -0,0 +1,427 @@
+ * [Register for post query events](register-for-post-query-events) (for a particular Oracle)
+ * [Register for query response events](register-for-query-response-events) (for a particular Oracle query)
+
+## General Request/Response types
+A request has the format:
+
+| Name | Type | Description | Required |
+| ---- | ---- | ----------- | -------- |
+| target | string | what component to target | Yes |
+| action | string | what is the action | Yes |
+| payload | object | data for action | Yes |
+
+A response (and an event) has the format:
+
+| Name | Type | Description | Required |
+| ---- | ---- | ----------- | -------- |
+| origin | string | what component did the action | Yes |
+| action | string | what was the action | Yes |
+| payload | object | data from the action | Yes |
+
+## [Register an Oracle]
+### Request
+ * **target:** `oracle`
+ * **action:** `register`
+ * **payload:**
+
+  | Name | Type | Description | Required |
+  | ---- | ---- | ----------- | -------- |
+  | type | `OracleRegisterTxObject` | | Yes |
+  | vsn | number | | No |
+  | query_format | string |  | Yes |
+  | response_format | string |  | Yes |
+  | query_fee | number |  | Yes |
+  | ttl | object - [TTL](#ttl) |  | Yes |
+  | fee | number |  | Yes |
+
+### Response
+ * **origin:** `oracle`
+ * **action:** `register`
+ * **payload:**
+
+  | Name | Type | Description | Required |
+  | ---- | ---- | ----------- | -------- |
+  | result | string | `ok` or error reason | Yes |
+  | oracle_id | string | Hash for created oracle | Yes on success |
+
+### Example
+```
+Request:
+  {"target":"oracle",
+   "action":"register",
+   "payload":{"type":"OracleRegisterTxObject",
+              "vsn":1,
+              "query_format":"the query spec",
+              "response_format":"the response spec",
+              "query_fee":4,
+              "ttl":{"type":"delta", "value":50},
+              "fee":5}
+  }
+
+Response:
+  {"action":"register",
+   "origin":"oracle",
+   "payload":{"result":"ok",
+              "oracle_id":"ok$3jzZyCLFtHVD7yVdEhGJFM3LjeXrKqWxnHbCYzhnrrR4DkdFtaJuxQvrR8VbbXExDPkCHFAei5q969JA6EayQpb8z5C3Mf"}
+  }
+```
+
+## Post a query for an Oracle
+### Request
+ * **target:** `oracle`
+ * **action:** `query`
+ * **payload:**
+
+  | Name | Type | Description | Required |
+  | ---- | ---- | ----------- | -------- |
+  | type | `OracleQueryTxObject` | | Yes |
+  | vsn | number | | No |
+  | oracle_pubkey | string |  | Yes |
+  | query | string |  | Yes |
+  | query_fee | number |  | Yes |
+  | query_ttl | object - [TTL](#ttl) |  | Yes |
+  | response_ttl | object - [RelativeTTL](#relativettl) |  | Yes |
+  | fee | number |  | Yes |
+
+### Response
+ * **origin:** `oracle`
+ * **action:** `query`
+ * **payload:**
+
+  | Name | Type | Description | Required |
+  | ---- | ---- | ----------- | -------- |
+  | result | string | `ok` or error reason | Yes |
+  | query_id | string | Hash for created query | Yes on success |
```

Comment:
Is the query_id an id per-oracle or across all oracles?

Reply:
and is user meant to be able to extract oracle from query (I think not needed)?

Reply:
query_id is unique (across all oracles) 

no, you can't extract the oracle_id from the query_id